### PR TITLE
Feat: test visualization

### PR DESCRIPTION
**Pull request by sphamba**
# Summary
- add tests directory to coverage
- wip: add tests for `plot.py`

### CODE REVIEW

File: .github/workflows/tests_windows.yml
Comment (ghiggi) [inline .github/workflows/tests_windows.yml:22]: Should also the other repo be updated?

File: gpm_api/tests/test_visualization/test_plot.py
Comment (ghiggi) [inline gpm_api/tests/test_visualization/test_plot.py:36]: I wonder if it's not better to pass the plotter object (or the figure object) returned by the plotting functions as argument to this function... and then do `plotter.figure.savefig()` or  `fig.save_fig()`. This would avoid searching in the global environment for the plot.

Example: 

```
p = plot.plot_map_mesh(grid_antimeridian_dataarray)
 save_and_check_figure(plotter=p, name=get_test_name(self))
```
or
```
p = plot.plot_map_mesh(grid_antimeridian_dataarray)
 save_and_check_figure(fig=p.figure, name=get_test_name(self))
```

**Review by ghiggi — APPROVED** [review]
Looks very clean. 
You should document what must be done to get the data from the `gpm_api_test_data` repository! 

Regarding the plots: 
- for https://github.com/ghiggi/gpm_api_test_data/blob/main/plots/test_plot-TestPlotMap-test_orbit_antimeridian.png we might want to automatically recenter the projection when an orbit cross a single time the antimeridian (but not the poles)
- For the plot_map_mesh of orbit intersecting the antimeridian (https://github.com/ghiggi/gpm_api_test_data/blob/main/plots/test_plot-TestPlotMapMesh-test_orbit_antimeridian.png) and the pole (https://github.com/ghiggi/gpm_api_test_data/blob/main/plots/test_plot-TestPlotMapMesh-test_orbit_pole_projection.png)... I can have a look to implement a fix ;)

File: gpm_api/tests/test_visualization/test_plot.py
Comment (ghiggi) [inline gpm_api/tests/test_visualization/test_plot.py:381]: The bad results in the picture arise from the setting of the argument `edgecolors`, because the cell lines crossing the antimeridian (projection boundary) are not masked (only cell facecolors are masked). 
We could open an issue in cartopy to see if the maintainers foresee an easy solution to also mask the cell edges crossing the projection boundaries.

I don't see a straightforward solution in GPM-API for this case: 
- if the mesh to be plot is local (a small region crossing the antimeridian) we could detect when crossing the antimeridian only once and modify the CRS if is set to PlateeCarre projection and center it to 180 instead of 0. 
- But if the mesh to be plot is of multiple orbits around the earth, the only solution is to call the cartopy pcolormesh function only over orbit along-track slices where any cell intersects the antimeridian.

File: gpm_api/tests/test_visualization/test_plot.py
Comment (ghiggi) [inline gpm_api/tests/test_visualization/test_plot.py:415]: Why centered on -30? If SouthPole should be `ccrs.Orthographic(180, -90)` no? 
Or alternatively `ccrs.SouthPolarStereo()`

**Review by ghiggi — COMMENTED** [review]
:+1:

File: gpm_api/tests/test_visualization/test_plot.py
Comment (sphamba) [reply gpm_api/tests/test_visualization/test_plot.py:36]: Yes that is better, less bug prone

**Comment by sphamba** [comment]
We should open issues for your comments related to changes in the code (not the tests).
I will tackle 3d and transect in another PR

File: gpm_api/tests/test_visualization/test_facetgrid.py
Comment (ghiggi) [inline gpm_api/tests/test_visualization/test_facetgrid.py:161]: I think there is not a test unit for a plot_map or plot_image (without facetgrid) with a horizontal colorbar !

File: gpm_api/tests/test_visualization/test_facetgrid.py
Comment (ghiggi) [inline gpm_api/tests/test_visualization/test_facetgrid.py:191]: To remove. Cartopy does not allow aspect ratio different than 1. And optimize_layout destroy everything.
But `facet_aspect` works correctly for `plot_image` to derive `facet_width` based on the specified `faced_height` f I recall correctly ;)

File: gpm_api/tests/test_visualization/test_plot.py
Comment (ghiggi) [inline gpm_api/tests/test_visualization/test_plot.py:245]: Test also rgba ;)

Comment (ghiggi) [reply gpm_api/tests/test_visualization/test_plot.py:245]: And test also with nan in the lat/lon coordinates

File: gpm_api/tests/test_visualization/test_plot.py
Comment (ghiggi) [reply gpm_api/tests/test_visualization/test_plot.py:397]: I think we can archive this comment. 
Anything can't be done except a fix in cartopy. We can open a issue there in case

**Review by ghiggi — COMMENTED** [review]
Few comments ;) 
Almost ready to be merged. 
I work now to the optmize_layout utility for plot_map

File: gpm_api/tests/test_visualization/test_plot.py
Comment (ghiggi) [inline gpm_api/tests/test_visualization/test_plot.py:216]: The figure from this test looks bad. 
But in the definition of  `orbit_nan_lon_cross_track_dataarray`, you used `cross_track_index = 1` which is not the outer cross-track ... if you use `cross_track_index = 0` the bad behaviour does not occur I guess. Can you try out? 
If the figures looks correct, please rename this test as `test_orbit_invalid_outer_cross_track_coordinates` (for `cross_track_index = 0`)

But I guess you exposed a bug in the interpolation of invalid coordinates ... so can you open an issue with a reproducible example of this plot? I can try to fix it tomorrow or Friday after this big PR is merged.

**Review by ghiggi — COMMENTED** [review]
@sphamba I added a couple of explanations for some stuffs that looks unclear. 
I let you address the comments and then we can merge the PR.

File: gpm_api/tests/test_visualization/test_plot.py
Comment (sphamba) [reply gpm_api/tests/test_visualization/test_plot.py:234]: So I made `test_orbit_nan_lon_one_cell`, `test_orbit_nan_lon_one_cell_centerline`, and `test_orbit_nan_lon_along_track`. For all cases, a whole line is missing. I suspect this has to do with the detection of continuous regions along-track.

File: gpm_api/tests/test_visualization/test_plot.py
Comment (sphamba) [reply gpm_api/tests/test_visualization/test_plot.py:216]: #41

**Review by ghiggi — APPROVED** [review]
LGTM. I merge ;)

**Pull request merged**
